hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -1747,6 +1747,26 @@ func (r *HostedClusterReconciler) reconcile(ctx context.Context, req ctrl.Reques
 		}
 	}
 
+	// Reconcile management infrastructure annotation
+	mgmtInfraKey := client.ObjectKey{Name: "cluster"}
+	mgmtInfra := &configv1.Infrastructure{}
+
+	if err := r.Get(ctx, mgmtInfraKey, mgmtInfra); err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to get infrastructure.config.openshift.io status: %w", err)
+	}
+	mgmtPlatformType := mgmtInfra.Status.PlatformStatus.Type
+
+	if _, exists := hcluster.Annotations[hyperv1.ManagementPlatformAnnotation]; !exists {
+		if hcluster.Annotations == nil {
+			hcluster.Annotations = map[string]string{}
+		}
+		hcluster.Annotations[hyperv1.ManagementPlatformAnnotation] = string(mgmtPlatformType)
+		if err := r.Client.Update(ctx, hcluster); err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to update hostedcluster %s annotation: %w", hcluster.Name, err)
+		}
+		log.Info(fmt.Sprintf("%s annotation set on the hostedcluster %s", hyperv1.ManagementPlatformAnnotation, hcluster.Name))
+	}
+
```

Comment:
since this is specifically used only for KubeVirt (for now), i think we should move this logic to the `reconcileKubevirtPlatformDefaultSettings` function. that way we don't potentially impact the other platforms. especially since this is somewhat "experimental" behavior we're exploring here for kubevirt on azure mgmt. 

if the `agent` platform wants to adopt this azure behavior in the future, they would add the mgmt cluster type annotation defaulting to their platform as well. so it would be an "opt in" type of behavior.

Reply:
relocated.